pkg/keyspace/tso_keyspace_group.go

```diff
@@ -219,17 +223,28 @@ func (m *GroupManager) watchServiceAddrs(ctx context.Context, revision int64) (i
 				return revision, wresp.Err()
 			}
 			for _, event := range wresp.Events {
-				s := &discovery.ServiceRegistryEntry{}
-				if err := json.Unmarshal(event.Kv.Value, s); err != nil {
```

Comment:
for delete event, json.Unmarshal(event.Kv.Value, s) will result in "unexpected end of JSON input" because of nil value. so it has to move to switch-case clientv3.EventTypePut: